src/App.java

```diff
@@ -1,5 +1,595 @@
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
 public class App {
+    private Library myLibrary;
+    private Printer printer = new Printer();
+
+    // ANSI color codes
+    private static final String WHITE = "\u001B[37m";
+    private static final String RED = "\u001B[31m";
+    private static final String GREEN = "\u001B[32m";
+    private static final String YELLOW = "\u001B[33m";
+    private static final String CYAN = "\u001B[36m";
+    private static final String LIGHT_BLUE = "\u001B[94m";
+
+    public void start() {
+        myLibrary = readLibraryData("library.json");
+        displayTitle();
+        displayMenu();
+    }
+
+    /**
+     * Reads the library data from a JSON file and populates the Library object (books and students array).
+     * @param filename
+     * @return
+     */
+    public Library readLibraryData(String filename) {
+        Library library = new Library();
+    
+        try {
+            printer.openFile(filename);
+    
+            String str = "";
+            while (printer.fileHasNextLine()) {
+                str = str + printer.fileGetNextLine() + "\n";
+            }
+            printer.closeFile();
+            String json = str.toString();
+    
+            Pattern bookPattern = Pattern.compile(

```

Comment:
Don't worry, I'm an old regex head. But this seems like a lot of work - did you look into JSON parsing libraries? they definitely exist, though I suppose for these relatively simple needs you don't need the flexibility of a full parser

Reply:
another option would be to look into the various options for java serialization, which allow for a more native way of saving and restoring java objects from files. the big disadvantage of them for this purpose is that they are not human-readable (but I would argue that manually editing a JSON file isn't a walk in the park for most normies either)